changelog/unreleased/issue-3456

```diff
@@ -0,0 +1,7 @@
+Enhancement: match untagged snapshots only when listing or forgetting
+
+When filtering snapshots, an empty tag `--tag ""` now matches all snapshots without tags.
```

Comment:
Unless @MichaelEischer thinks otherwise, I suggest changing the "" (double quotes) to '' (single quotes) since '' (single quotes) is what's being used in the examples in the documentation changes in this PR.